spring-session/src/main/java/org/springframework/session/data/gemfire/GemFireOperationsSessionRepository.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.session.data.gemfire;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.data.gemfire.GemfireOperations;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.Session;
+import org.springframework.util.Assert;
+
+import com.gemstone.gemfire.cache.query.SelectResults;
+import com.gemstone.gemfire.pdx.PdxReader;
+import com.gemstone.gemfire.pdx.PdxSerializable;
+import com.gemstone.gemfire.pdx.PdxWriter;
+
+/**
+ * The GemFireOperationsSessionRepository class is a Spring SessionRepository implementation that interfaces with
+ * and uses GemFire to back and store Spring Sessions.
+ *
+ * @author John Blum
+ * @see org.springframework.data.gemfire.GemfireOperations
+ * @see org.springframework.session.ExpiringSession
+ * @see org.springframework.session.Session
+ * @see org.springframework.session.data.gemfire.AbstractGemFireOperationsSessionRepository
+ * @since 1.1.0
+ */
+@SuppressWarnings("all")
+public class GemFireOperationsSessionRepository extends AbstractGemFireOperationsSessionRepository {
+
+	// GemFire OQL query used to look up Sessions by principal name.
+	protected static final String FIND_SESSIONS_BY_PRINCIPAL_NAME_QUERY =
```

Comment:
A note that we will need to update how we do the principal when we fix #301. The reason is that I think we will be making it so that any attribute can be indexed, so interface for finding by a session attribute will be more generic. Perhaps something like `findBySessionAttr(String name, String value)`